signup-service/src/main/java/io/mosip/signup/services/WebSocketHandler.java

```diff
@@ -60,25 +58,36 @@ public class WebSocketHandler {
 
 
     public void processFrames(IdentityVerificationRequest identityVerificationRequest) {
-        IdentityVerificationTransaction transaction = cacheUtilService.getVerifiedSlotTransaction(identityVerificationRequest.getSlotId());
-        if(transaction == null)
-            throw new InvalidTransactionException();
-
-        IdentityVerifierPlugin plugin = identityVerifierFactory.getIdentityVerifier(transaction.getVerifierId());
-        if(plugin == null)
-            throw new SignUpException(PLUGIN_NOT_FOUND);
-
-        if(plugin.isStartStep(identityVerificationRequest.getStepCode())) {
-            IdentityVerificationInitDto identityVerificationInitDto = new IdentityVerificationInitDto();
-            identityVerificationInitDto.setIndividualId(transaction.getIndividualId());
-            identityVerificationInitDto.setDisabilityType(transaction.getDisabilityType());
-            plugin.initialize(identityVerificationRequest.getSlotId(), identityVerificationInitDto);
-        }
+        String errorCode = null;
+        try {
+            validate(identityVerificationRequest);
+            IdentityVerificationTransaction transaction = cacheUtilService.getVerifiedSlotTransaction(identityVerificationRequest.getSlotId());
+            if(transaction == null)
+                throw new InvalidTransactionException();
+
+            IdentityVerifierPlugin plugin = identityVerifierFactory.getIdentityVerifier(transaction.getVerifierId());
+            if(plugin == null)
+                throw new SignUpException(PLUGIN_NOT_FOUND);
+
+            if(plugin.isStartStep(identityVerificationRequest.getStepCode())) {
+                IdentityVerificationInitDto identityVerificationInitDto = new IdentityVerificationInitDto();
+                identityVerificationInitDto.setIndividualId(transaction.getIndividualId());
+                identityVerificationInitDto.setDisabilityType(transaction.getDisabilityType());
+                plugin.initialize(identityVerificationRequest.getSlotId(), identityVerificationInitDto);
+            }
 
-        IdentityVerificationDto dto = new IdentityVerificationDto();
-        dto.setStepCode(identityVerificationRequest.getStepCode());
-        dto.setFrames(identityVerificationRequest.getFrames());
-        plugin.verify(identityVerificationRequest.getSlotId(), dto);
+            IdentityVerificationDto dto = new IdentityVerificationDto();
+            dto.setStepCode(identityVerificationRequest.getStepCode());
+            dto.setFrames(identityVerificationRequest.getFrames());
+            plugin.verify(identityVerificationRequest.getSlotId(), dto);
+        } catch (SignUpException e) {
+            errorCode = e.getErrorCode();
+            throw e;
+        } finally {
+            if (errorCode != null) {
+                simpMessagingTemplate.convertAndSend("/topic/" + identityVerificationRequest.getSlotId(), errorCode);
```

Comment:
We cannot directly return the errorCode @pvsaidurga, we should return an instance of "IdentityVerificationResult", and the feedback should be set with the type set to ERROR and the code set to actual errorCode.